soc: stm32: Add STM32L496XE support

gon1332 opened the pull request:
Added support for the [STM32L496XE series](https://www.st.com/en/microcontrollers-microprocessors/stm32l496re.html) with 512KB internal flash and 256KB SRAM of contiguous addresses.


Signed-off-by: Ioannis Konstantelias <ikonstadel@gmail.com>

gon1332 committed:
soc: stm32: Add STM32L496XE support

Added support for the STM32L496XE series with 512KB internal flash and
256KB SRAM of contiguous addresses.

Signed-off-by: Ioannis Konstantelias <ikonstadel@gmail.com>

gon1332 commented:
@nikooiko @galazari

psidhu commented:
> 512MB

That's a lot of space :smile:

gon1332 commented:
@psidhu Oops. Thanks. That's a hell of an upgrade. I've changed the PR description. I'll change the commit message as soon as possible.

gon1332 commented:
@psidhu @erwango I fixed the commit message. Could you check?

psidhu commented:
@gon1332 Commit message LGTM.

erwango commented on dts/arm/st/l4/stm32l496Xe.dtsi:
```diff
@@ -0,0 +1,21 @@
+
+/ {
+	sram0: memory@20000000 {
+		reg = <0x20000000 DT_SIZE_K(320)>;
```
SRAM size is actually 256KB.
We need to take into account that 64Kb RAM are located into SRAM2 at a non contiguous address.

gon1332 replied:
@erwango Done.

gon1332 commented:
@erwango I'd like to know if this PR is valid for Zephyr 1.14. Should I propose it now since it's not yet merged or on a new PR?

erwango commented:
> @erwango I'd like to know if this PR is valid for Zephyr 1.14. Should I propose it now since it's not yet merged or on a new PR?

@gon1332, I don't think it is. Aim is to fix bugs, not to add new stuff (new SoC in this case), but we can ask @nashif.

ioannisg commented:
Is there going to be a board for this SOC in the tree, @gon1332 ?

gon1332 commented:
@ioannisg AFAIC, there is no commercial board available. We only have our custom boards based on this SOC. So the answer is no, until there is something commercial.

erwango commented:
@gon1332 , due to use of `CONFIG_SOC_STM32L496XG` in https://github.com/zephyrproject-rtos/zephyr/pull/16813, I think you'll need to move `SOC_STM32L496XG` to `SOC_STM32L496XX`
so it works for both variants.
Change is safe, more details in https://github.com/zephyrproject-rtos/zephyr/issues/16733.

gon1332 committed:
soc: stm32: Remove package digit for STM32L496XG

Letter G is no longer needed as described in issue #16733.

Signed-off-by: Ioannis Konstantelias <ikonstadel@gmail.com>

gon1332 committed:
boards: Change to new STM32L496XX symbol

stm32l496zg_disco and nucleo_l496zg were depending on STM32L496XG SOC.
With the recent change to STM32L496XX, the dependency should be updated.

Signed-off-by: Ioannis Konstantelias <ikonstadel@gmail.com>

gon1332 commented:
@erwango Done
